src/firefly/java/edu/caltech/ipac/astro/FITSTableReader.java

```diff
@@ -193,70 +192,87 @@ else if (strategy == FULLY_FLATTEN) {
                                                          String[] headerCols,
                                                          String strategy)
         throws FitsException, IOException {
-
         List tList = getStarTableList(fits_filename);
 
         List<DataGroup> dgListTotal = new ArrayList();
-        // for testing:
-        //for (int i = 4; i < tList.size(); i++) {
-        for (int i = 0; i < tList.size(); i++) {
-            StarTable table = (StarTable) tList.get(i);
-            DataGroup dataGroup = convertStarTableToDataGroup(table, dataCols, headerCols, strategy);
+
+        for (Object table : tList) {
+            DataGroup dataGroup = convertStarTableToDataGroup((StarTable)table, dataCols, headerCols, strategy);
             dgListTotal.add(dataGroup);
         }
         return dgListTotal;
     }
 
     /**
-     * Get a list of star tables from a FITS file.
+     * Convert a table from a FITS file to DataGroup based on table index
      * @param fits_filename
-     * @return List of star tables.
+     * @param dataCols: The names of the columns which will be copied to the data section of the DataGroups.
+     *                If dataCols = null, get all the columns into the data group.
+     * @param headerCols: The names of the columns which will be copied to the header section of the DataGroups.
+     *                If headerCols = null, get none of the columns into the header of the data group.
+     * @param strategy The strategy used to deal with the repeat count  of the data in the given dataCols columns.
+     * @param table_idx
+     * @return
+     * @throws FitsException
      * @throws IOException
      */
-    public static List getStarTableList(String fits_filename) {
+    public static DataGroup convertFitsToDataGroup(String fits_filename,
+                                                         String[] dataCols,
+                                                         String[] headerCols,
+                                                         String strategy, int table_idx)
+            throws FitsException, IOException {
+
+        StarTable table = getStarTableList(fits_filename, table_idx);
+        return table != null ? convertStarTableToDataGroup(table, dataCols, headerCols, strategy) : null;
+    }
+
+    /**
+     * Get a star tables from a FITS file.
+     * @param fits_filename
+     * @param table_idx
+     * @return one star table or none
+     * @throws IOException
+     */
+
+    public static StarTable getStarTableList(String fits_filename, int table_idx) {
```

Comment:
Really minor.  Will you rename the method to getStarTable?  It is not returning a list anymore.
Feel free to merge after your change.